src/include/platform/internal/GenericPlatformManagerImpl.h

```diff
@@ -57,7 +57,10 @@ class GenericPlatformManagerImpl
     void _RemoveEventHandler(PlatformManager::EventHandlerFunct handler, intptr_t arg);
     void _HandleServerStarted();
     void _HandleServerShuttingDown();
-    void _ScheduleWork(AsyncWorkFunct workFunct, intptr_t arg);
+    CHIP_ERROR _ScheduleWork(AsyncWorkFunct workFunct, intptr_t arg);
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING) && CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+    CHIP_ERROR _ScheduleBackgroundWork(AsyncWorkFunct workFunct, intptr_t arg);
+#endif
```

Comment:
This API should not be conditional. It should always be there, but it should implement internally to just ScheduleWork if CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING not set

Reply:
I've changed it so the functions are always present, and CASE Sigma3 always does a/b/c steps in fg/bg/fg, except that if the BG processing feature is disabled, it falls back to FG. (This can still increase liveness in the event loop, incidentally.)